lightning/src/ln/channelmanager.rs

```diff
@@ -333,29 +337,173 @@ mod inbound_payment {
 		}
 		return Ok(PaymentPreimage(decoded_payment_preimage))
 	}
+}
+
+/// LDK has multiple reasons to generate fake short channel ids:
+/// 1) zero-conf channels that don't have a confirmed channel id yet
+/// 2) phantom node payments, to get an scid for the phantom node's phantom channel
+mod fake_scid {
+	use bitcoin::hash_types::BlockHash;
+	use bitcoin::hashes::hex::FromHex;
+	use chain::keysinterface::{Sign, KeysInterface};
+	use util::scid_utils;
+
+	use core::convert::TryInto;
+	use core::ops::Deref;
+
+	const TEST_SEGWIT_ACTIVATION_HEIGHT: u32 = 0;
+	const MAINNET_SEGWIT_ACTIVATION_HEIGHT: u32 = 481_824;
+	const MAX_TX_INDEX: u32 = 2_500;
+	const MAX_VOUT: u16 = 3;
+	const MAX_NAMESPACES: u8 = 8;
+	const NAMESPACE_ID_BITMASK: u8 = 0b111;
+
+	/// Fake scids are divided into namespaces, with each namespace having its own randomly-selected
+	/// identifier between [0..7], such that `scid_tx_index % MAX_NAMESPACES == namespace_identifier`.
+	/// This allows us to identify what namespace a fake scid corresponds to upon HTLC receipt, and
+	/// handle the HTLC accordingly.
+	pub(super) enum Namespace {
+		Phantom(u8),
+		// Coming soon: a variant for the zero-conf scid namespace
+	}
+
+	impl Namespace {
+		/// We generate "realistic-looking" random scids here, meaning the scid's block height is
+		/// between segwit activation and the current best known height, and the tx index and output
+		/// index are also selected from a "reasonable" range. We add this logic because it makes it
+		/// non-obvious at a glance that the scid is fake, e.g. if it appears in invoice route hints.
+		pub(super) fn get_fake_scid<Signer: Sign, K: Deref>(&self, highest_seen_blockheight: u32, genesis_hash: &BlockHash, keys_manager: &K) -> u64
+			where K::Target: KeysInterface<Signer = Signer>,
+		{
+			const BLOCKS_PER_DAY: u32 = 144;
+			let scid_namespace = self.as_u8();
+			let rand_bytes = keys_manager.get_secure_random_bytes();
+
+			let segwit_activation_height = segwit_activation_height(genesis_hash);
+			let mut valid_block_range = if highest_seen_blockheight > segwit_activation_height {
+				highest_seen_blockheight - segwit_activation_height
+			} else {
+				1
+			};
+			// We want to ensure that this fake channel won't conflict with any transactions we haven't
+			// seen yet, in case `highest_seen_blockheight` is updated before we get full information
+			// about transactions confirmed in the given block.
+			if valid_block_range > BLOCKS_PER_DAY { valid_block_range -= BLOCKS_PER_DAY; }
```

Comment:
Actually, can we make the constant here much bigger? I'm worried somewhat about legitimate collisions as-is, and while we could write a ton of code to handle collisions, we could also just tell users "you're not allowed to ever be behind by a full month and waiting for a channel to confirm"? Then we can just `expect` and let users figure it out if things crash on a collision?

Reply:
Parsing this change -- I'm fine to change `BLOCKS_PER_DAY` to `BLOCKS_PER_MONTH`. But, where are you saying this `expect` would go? Also, to confirm, I'd think we still need to check that `valid_block_range > BLOCKS_PER_*` before subtracting here, since in testing there may not be enough block height.

Reply:
Yea, sorry, a bit of context was missing - when we do scid aliases we'll want to track mappings from scid to real channel, and there we will want to assert on collisions with real channels. we wont need them here.

Reply:
Changed the constant. Can't find much information on scid aliases or their motivation, let me know if I can get more context here

Reply:
SCIDAliases == the alias we use for a channel when its active after 0 conf.